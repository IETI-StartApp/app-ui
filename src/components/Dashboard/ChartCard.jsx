import React from 'react';
import { useStyles } from './styles';
import { Card, CardContent, Typography, Box } from '@material-ui/core/';

export const ChartCard = ({ title, chart}) => {
    const classes = useStyles();
    
    return (
        <Card variant="outlined" className={classes.cardChart}>
            <CardContent>
                <Typography className={classes.mainColor} variant="button" display="block">{ title }</Typography>
                <Box p={3}>
                    {
                        chart
                    }
                </Box>
            </CardContent>
        </Card>
    );
}