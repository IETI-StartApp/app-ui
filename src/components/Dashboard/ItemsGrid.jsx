import React from 'react';
import { Card, CardContent, Typography, Avatar, Grid, CardActionArea } from '@material-ui/core/';
import { useStyles } from './styles';
import { useHistory } from "react-router";

export const ItemsGrid = ({ items }) => {
    const classes = useStyles();
    const history = useHistory() || [];

    return (
        <Grid container spacing={2} direction="row" justify="center" className={classes.items}>
            {
                items.map(item => (
                    <Grid item xs={12} sm={5} lg={3}>
                        <div style={{ display: "flex", alignItems: "center", justifyContent: "center" }}>
                            <Card variant="outlined" className={classes.dashItem}>
                                <Avatar className={classes.cardAvatar} variant="rounded">
                                    {item.icon}
                                </Avatar>
                                <CardActionArea onClick={() => { history.push(item.path) }}>
                                    <CardContent className={classes.itemContent}>
                                        <Typography variant="h4" align="center" className={classes.mainColor} >{item.number}</Typography>
                                        <Typography align="center">{item.description}</Typography>
                                    </CardContent>
                                </CardActionArea>
                            </Card>
                        </div>
                    </Grid>
                ))
            }
        </Grid>
    );
}