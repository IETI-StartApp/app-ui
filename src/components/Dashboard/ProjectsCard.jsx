import React from 'react';
import { Card, CardContent, Typography, GridList, GridListTile } from '@material-ui/core/';
import CardProject from '../ProjectItems/CardProject';
import { useStyles } from './styles';

export const ProjectsCard = ({ title, projects }) => {
    const classes = useStyles();

    return (
        <Card variant="outlined" className={classes.projects}>
            <CardContent>
                <Typography className={classes.mainColor} variant="button" display="block">{ title }</Typography>
                <div className={classes.rootProjects}>
                    <GridList className={classes.gridList}>
                        {projects.map((project) => (
                            <GridListTile className={classes.gridListTile}>
                                <CardProject project={project} />
                            </GridListTile>
                        ))}
                    </GridList>
                </div>
            </CardContent>
        </Card>
    );
}