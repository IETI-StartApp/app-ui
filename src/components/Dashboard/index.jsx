import React from 'react';
import Navigation from '../Navbar/Navigation';
import ChatBubbleOutlineIcon from '@material-ui/icons/ChatBubbleOutline';
import { Grid, Fab } from '@material-ui/core/';
import { useStyles } from './styles';
import { ItemsGrid } from './ItemsGrid';
import { ProjectsCard } from './ProjectsCard';
import { ChartCard } from './ChartCard';

export const Dashboard = ( { items, projects, chart, projectsTitle, chartTitle } ) => {
    const classes = useStyles();

    return (
        <>
            <Navigation />
            <div className={classes.offset}></div>
            <div className={classes.root}>
                <Grid container spacing={2} direction="col" justify="center" alignItems="center" className={classes.allItems}>
                    <Grid item xs={12}>
                        <ItemsGrid items={items} />
                    </Grid>
                    <Grid item xs={12}>
                        <Grid container spacing={2} direction="row" justify="center">
                            <Grid item xs={12} sm={10} lg={6}>
                                <ChartCard title= { chartTitle } chart = { chart } />
                            </Grid>
                            <Grid item xs={12} sm={10} lg={6}>
                                <ProjectsCard title={ projectsTitle } projects={ projects } />
                            </Grid>
                        </Grid>
                    </Grid>
                </Grid>
                <Fab aria-label="Chat" className={classes.fab}>
                    <ChatBubbleOutlineIcon />
                </Fab>
            </div>
        </>
    );
}