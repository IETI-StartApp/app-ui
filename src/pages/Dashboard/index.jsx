import React from 'react';
import { InvestorDashboard, EntrepreneurDashboard} from "../../components";
import { useLocation } from "react-router-dom";

export const Dashboard = () => {
    const {state : {role} = "ENTREPENEUR"} = useLocation();
    return (
        <>
            {
                role === "INVESTOR" &&
                <InvestorDashboard />
            }
            {
                role === "ENTREPRENEUR" &&
                <EntrepreneurDashboard />
            }
        </>
    );
}
