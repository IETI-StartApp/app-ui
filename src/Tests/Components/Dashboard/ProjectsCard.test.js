import React from 'react';
import { ProjectsCard } from '../../../components/Dashboard/ProjectsCard';
import { shallow } from 'enzyme';

describe('Testing projects card', () => {
    const projects = [
        {
            name: "Proyecto 1", 
            image:"https://www.binarias.org/wp-content/uploads/2020/12/tegnologia.jpeg", 
            finance: { value:15000000, valuation:20000000, startDate: "10/03/2021"}
        },
        {
            name: "Proyecto 2", 
            image:"https://s3-us-west-2.amazonaws.com/wp-mpro-blog/wp-content/uploads/2016/03/14133959/s3-blog-tecnologia-como-prioridad-min.png", 
            finance: { value:10000000, valuation:17500000, startDate: "12/02/2021"},
        },
        {
            name: "Proyecto 3", 
            image:"https://thelogisticsworld.com/wp-content/uploads/2020/02/empresario-analizando-informe-financiero-empresa-graficos-realidad-aumentada_34141-360.jpg", 
            finance: { value:8000000, valuation:12000000, startDate: "21/01/2021"},
        },
    ];

    const wrapper = shallow(<ProjectsCard  projects={ projects } title="projects" />)

    test('should projects card', () => {
        expect(wrapper).toMatchSnapshot();
    });

})