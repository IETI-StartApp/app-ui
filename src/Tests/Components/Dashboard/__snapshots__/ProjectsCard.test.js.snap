// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing projects card should projects card 1`] = `
<WithStyles(ForwardRef(Card))
  className="makeStyles-projects-6"
  variant="outlined"
>
  <WithStyles(ForwardRef(CardContent))>
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-mainColor-7"
      display="block"
      variant="button"
    >
      projects
    </WithStyles(ForwardRef(Typography))>
    <div
      className="makeStyles-rootProjects-9"
    >
      <WithStyles(ForwardRef(GridList))
        className="makeStyles-gridList-10"
      >
        <WithStyles(ForwardRef(GridListTile))
          className="makeStyles-gridListTile-11"
        >
          <CardProject
            project={
              Object {
                "finance": Object {
                  "startDate": "10/03/2021",
                  "valuation": 20000000,
                  "value": 15000000,
                },
                "image": "https://www.binarias.org/wp-content/uploads/2020/12/tegnologia.jpeg",
                "name": "Proyecto 1",
              }
            }
          />
        </WithStyles(ForwardRef(GridListTile))>
        <WithStyles(ForwardRef(GridListTile))
          className="makeStyles-gridListTile-11"
        >
          <CardProject
            project={
              Object {
                "finance": Object {
                  "startDate": "12/02/2021",
                  "valuation": 17500000,
                  "value": 10000000,
                },
                "image": "https://s3-us-west-2.amazonaws.com/wp-mpro-blog/wp-content/uploads/2016/03/14133959/s3-blog-tecnologia-como-prioridad-min.png",
                "name": "Proyecto 2",
              }
            }
          />
        </WithStyles(ForwardRef(GridListTile))>
        <WithStyles(ForwardRef(GridListTile))
          className="makeStyles-gridListTile-11"
        >
          <CardProject
            project={
              Object {
                "finance": Object {
                  "startDate": "21/01/2021",
                  "valuation": 12000000,
                  "value": 8000000,
                },
                "image": "https://thelogisticsworld.com/wp-content/uploads/2020/02/empresario-analizando-informe-financiero-empresa-graficos-realidad-aumentada_34141-360.jpg",
                "name": "Proyecto 3",
              }
            }
          />
        </WithStyles(ForwardRef(GridListTile))>
      </WithStyles(ForwardRef(GridList))>
    </div>
  </WithStyles(ForwardRef(CardContent))>
</WithStyles(ForwardRef(Card))>
`;
